### ISSUE TITLE
overflowed its stack

### ISSUE THREAD

**Issue by yinheli** [equalitie/btdht#7]
thread 'tokio-runtime-worker' has overflowed its stack
fatal runtime error: stack overflow
Aborted

https://github.com/equalitie/btdht/blob/master/src/worker/socket.rs#L53

recursively calling self?

```rust
#[async_trait]
impl SocketTrait for UdpSocket {
    async fn send_to(&self, buf: &[u8], target: &SocketAddr) -> io::Result<()> {
        self.send_to(buf, target).await.map(|_| ())
    }

    async fn recv_from(&mut self, buf: &mut [u8]) -> io::Result<(usize, SocketAddr)> {
        self.recv_from(buf).await
    }

    fn local_addr(&self) -> io::Result<SocketAddr> {
        self.local_addr()
    }
}
```

**Comment by inetic** [comment]
> recursively calling self?

I think you're right, but it's weird because we use it [here](https://github.com/equalitie/btdht/blob/6b4246b10be4bbebf2bcd9ac5933ad74355bb9b0/src/worker/socket.rs#L46-L59) similarly and there it works as expected. Could you try what Stack Overflow suggests [here](https://stackoverflow.com/a/63935634)?

That is:

```rust
#[async_trait]
impl SocketTrait for UdpSocket {
    async fn send_to(&self, buf: &[u8], target: &SocketAddr) -> io::Result<()> {
        UdpSocket::send_to(buf, target).await.map(|_| ())
    }

    async fn recv_from(&mut self, buf: &mut [u8]) -> io::Result<(usize, SocketAddr)> {
        UdpSocket::recv_from(buf).await
    }

    fn local_addr(&self) -> io::Result<SocketAddr> {
        UdpSocket::local_addr()
    }
}
```

**Comment by inetic** [comment]
It should now be fixed by 5287b4a2bc507cde87903bc30ca2f1c548a31e18, thanks for reporting.
